go-controller/extensions/pkg/node/node_windows.go

```diff
@@ -40,4 +135,84 @@ func (n *nodeController) Delete(node *kapi.Node) {
 //  1. Setting up this node and its VxLAN extension for talking to other nodes
 //  2. Setting back annotations about its VTEP and gateway MAC address to its own node object
 func (n *nodeController) InitSelf() {
+	// Create network
+	var networkInfo NetworkInfo
+
+	// Retrieve the host prefix from the annotations
+	hostsubnet, ok := n.thisNode.Annotations[extensionHostSubnet]
+
+	if !ok {
+		log.Errorf("Couldn't retreive the host subnet from the '%s' node's annotations", n.thisNode.Name)
+		return
+	}
+
+	_, addressPrefix, err := net.ParseCIDR(hostsubnet)
+	if err != nil {
+		log.Error(err)
+		return
+	}
+
+	// Retrieve the host prefix from the annotations
+	gatewayIP, ok := n.thisNode.Annotations[extensionDrIp]
+
+	if !ok {
+		log.Errorf("Couldn't retreive the gateway IP address from the '%s' node's annotations", n.thisNode.Name)
+		return
+	}
+
+	gatewayAddress := net.ParseIP(gatewayIP)
+	networkInfo.AutomaticDNS = useAutomaticDns
+	networkInfo.InterfaceName = gatewayInterfaceName
+	networkInfo.Name = networkName
+
+	networkInfo.Subnets = []SubnetInfo{
+		SubnetInfo{
+			AddressPrefix:  *addressPrefix,
+			GatewayAddress: gatewayAddress,
+		},
+	}
+
+	networkSchema, err := networkInfo.GetHostComputeNetworkConfig()
+	if err != nil {
+		log.Error("Unable to create the overlay network, error: %v", err)
+		return
+	}
+
+	network, err := networkSchema.Create()
```

Comment:
@JocelynBerrendonner is this operation idempotent? eg if the node object somehow goes away and is re-created, will this call fail because the schema is already created? I'm not quite sure which operations are local and which are remote with HNS :)

Reply:
Based on our discussion, I completely reworked the flow. I am basically doing all the node configuration within "add" and "delete". In these calls, I special case the local node to create an HNS network, and configure it. Other nodes are just dealing with configuring VXLAN tunnels.